packages/blocks/src/api/utils.js

```diff
@@ -121,3 +122,31 @@ export function normalizeBlockType( blockTypeOrName ) {
 
 	return blockTypeOrName;
 }
+
+/**
+ * Get the block label, using the block title and display name if it has one.
+ * Use only the block title if it doesn't have a display name.
+ *
+ * @param {Object}  blockType  The block type.
+ * @param {Object}  attributes The values of the block's attributes.
+ * @param {?string} separator  A separator to display between the title and
+ *                             displayName. Defaults to a colon (': ').
+ *
+ * @return {string} The block label.
+ */
+export function getBlockLabel( blockType, attributes, separator = ': ' ) {
+	const {
+		__experimentalDisplayName: displayNameAttribute,
+		title: blockTitle,
+	} = blockType;
+
+	if ( ! displayNameAttribute || ! attributes[ displayNameAttribute ] ) {
+		return blockTitle;
+	}
+
+	// Strip any formatting.
+	const richTextValue = create( { html: attributes[ displayNameAttribute ] } );
+	const formatlessDisplayName = getTextContent( richTextValue );
+
+	return `${ blockTitle }${ separator }${ formatlessDisplayName }`;
```

Comment:
What if I wanted to avoid using the block title and separator format? Currently, we are working on some blocks that look like this:

Before:
![Screen Shot 2019-10-28 at 6 34 54 PM](https://user-images.githubusercontent.com/28742426/67723213-b86a1b80-f9b1-11e9-86a6-83874b210686.png)

After:
![Screen Shot 2019-10-28 at 6 34 10 PM](https://user-images.githubusercontent.com/28742426/67723221-bd2ecf80-f9b1-11e9-99ed-86d1162a4c60.png)

I think we'd like to just use "Header" instead of displaying "Template: Header". To us, the "Template" block (Which is dubiously named. One shouldn't trust what they think "template" means! 😆) is just a transparent wrapper around some other blocks and isn't really meant to be understood by or visible to the user.

Reply:
Good to know. A previous proposal has been to use a function or component to create the label, which is useful if the label is created from multiple attributes.
```
__experimentalDisplayName( { attributes, blockType } ) {
  return `${ blockType.title }: ${ attributes.label }`
}
```

My concern would be a lack of consistency across blocks. Also where this is used as an aria-label, the block type might make sense, whereas in the navigator it might not.

I suppose this could be customisable by `getBlockLabel`:
```
getBlockLabel( blockType, blockAttributes, { includeBlockTitle: false } );
```
So that aria-labels can include the block titles, but the navigator doesn't.

Reply:
How about the 3rd param is more general and contains params like:
```js
{
    context: 'block-navigation',
    format: 'concise',
}
```
Then each block's author could decide how to code the label depending on the params passed. We could always provide a default handler which works fine in 90% of cases.

Reply:
I'm still doubtful of the use case in general. We are not going to be using "Menu Item: Label" for navigation, for example. If we'd also not use it for template parts, we might be preemptively overcomplicating things here.

Reply:
I'm thinking the best approach might be to have two functions `getAccessibleBlockLabel` and `getVisualBlockLabel`. The former can include the block title for some extra context. Thoughts?

I've rebased and made the change described above, though it seems at the moment there's a bug with the nav block in `master`, so hard to test it.